### PR TITLE
Documentation

### PR DESCRIPTION
**Pull request by ranzmigrod**

### CODE REVIEW

File: src/DroneSafetyApi/Controllers/HeatmapsController.cs [hunk 1]
```diff
@@ -9,12 +9,30 @@
...
+    /// <summary>
+    /// The HeatmapsController class is a controller to recieve queries from and return <see cref="HeatmapsResponse"/>s  to
+    /// the web application
+    /// </summary>
     [Route("api/[controller]")]
     [EnableCors("CorsPolicy")]
     public class HeatmapsController : Controller
     {
+        /// <summary>
+        /// The Hazards property is used to query for a collection of <see cref="Hazard"/>s
```
Comment (dhruvtapasvi) [inline src/DroneSafetyApi/Controllers/HeatmapsController.cs]: "query the database"

File: src/DroneSafetyApi/Controllers/HeatmapsController.cs [hunk 2]
```diff
@@ -9,12 +9,30 @@
...
+        /// into <see cref="IHazardsToHeatmapsResponse"/>s
+        /// </summary>
         public IHazardsToHeatmapsResponse HazardsToHeatmaps { get; set; }
+        /// <summary>
+        /// Creates a new instance of the <see cref="HeatmapsController"/> class
+        /// </summary>
+        /// <param name="hazards"> The <see cref="IHazardRepository"/> used to query for <see cref="Hazards"/> in the new
+        /// <see cref="HeatmapsController"/> instance </param>
+        /// <param name="hazardsToHeatmaps"> The <see cref="IHazardsToHeatmapsResponse"/> used to generate <see cref="HeatmapsResponse"/>s
+        /// in the new <see cref="HeatmapsController"/> instanc</param>
```
Comment (dhruvtapasvi) [inline src/DroneSafetyApi/Controllers/HeatmapsController.cs]: instance spelling last word

File: src/DroneSafetyApi/Controllers/HeatmapsController.cs [hunk 3]
```diff
@@ -23,6 +41,11 @@ public class HeatmapsController : Controller
             HazardsToHeatmaps = hazardsToHeatmaps;
         }
 
+        /// <summary>
+        /// The GetHeatmaps method generates a <see cref="HeatmapsResponse"/> from a <see cref="HeatmapsQuery"/>
```
Comment (dhruvtapasvi) [inline src/DroneSafetyApi/Controllers/HeatmapsController.cs]: ...and serves it back as an HTTP response

File: src/DroneSafetyApi/Data/DbOptions.cs [hunk 4]
```diff
@@ -5,10 +5,22 @@
 
 namespace DroneSafetyApi.Data
 {
+    /// <summary>
+    /// The DbOptions class is a data model to represents configurations for a DocumentDb query
+    /// </summary>
     public class DbOptions
     {
+        /// <summary>
+        /// The EndpointUri property represents the the endpoint URI of the DocumentDB database
```
Comment (dhruvtapasvi) [inline src/DroneSafetyApi/Data/DbOptions.cs:14]: Do you have to specify "The EndpointUri property..." or can you just say "represents..."

File: src/DroneSafetyApi/Data/ExampleHazardRepository.cs [hunk 5]
```diff
@@ -9,7 +9,7 @@ namespace DroneSafetyApi.Data
 {
     public class ExampleHazardRepository : IHazardRepository
```
Comment (dhruvtapasvi) [inline src/DroneSafetyApi/Data/ExampleHazardRepository.cs]: Perhaps we can given a short explanation of what the examples repository does, only for the class

File: src/DroneSafetyApi/Models/Bounds.cs [hunk 6]
```diff
@@ -6,10 +6,24 @@
...
         public Position Min { get; private set; }
+        /// <summary>
+        /// The Max property represents the coordinates for the maximum Longitude and Latitude of the bounding box
+        /// </summary>
         public Position Max { get; private set; }
+        /// <summary>
+        /// Creates a new instance of the <see cref="Bounds"/> class 
+        /// </summary>
+        /// <param name="cornerOne">The first corner of the bounding box</param>
+        /// /// <param name="cornerTwo">The second corner of the bounding box</param>
```
Comment (dhruvtapasvi) [inline src/DroneSafetyApi/Models/Bounds.cs]: Why 2 of "///"?

File: src/DroneSafetyApi/Models/HeatMapPoint.cs [hunk 7]
```diff
@@ -1,9 +1,21 @@
...
+    /// The HeatmapPoint class is a model to hold the state of heatmap points that are to be sent to the web application
+    /// </summary>
     public class HeatmapPoint
     {
+        /// <summary>
+        /// The X property represents the Longitude coordinate of the heatmap point
+        /// </summary>
         public double X { get; set; }
+        /// <summary>
+        /// The XY property represents the Latitude coordinate of the heatmap point
```
Comment (dhruvtapasvi) [inline src/DroneSafetyApi/Models/HeatMapPoint.cs]: Should be "The Y property"?

File: src/DroneSafetyApi/Models/HeatmapsResponse.cs [hunk 8]
```diff
@@ -2,10 +2,22 @@
 
 namespace DroneSafetyApi.Models
 {
+    /// <summary>
+    /// The HeatmapResponse class is a model to hold information that is being sent to the web application
```
Comment (dhruvtapasvi) [inline src/DroneSafetyApi/Models/HeatmapsResponse.cs]: We shouldn't mention the web application. Would be better to say, "model to hold information served over the api"

File: src/DroneSafetyApi/Models/HeatmapsResponse.cs [hunk 9]
```diff
@@ -2,10 +2,22 @@
...
+        /// <summary>
+        /// The NumSources property represents the number of different data sources processed
+        /// </summary>
         public int NumSources { get; set; }
+        /// <summary>
+        /// The Sources property represents the collection of names of the different data sources processed
+        /// </summary>
         public IEnumerable<string> Sources { get; set; }
+        /// <summary>
+        /// The Heatmaps property represents a mapping from a data source to the <see cref="HeatmapPoint"/>s that display it
```
Comment (dhruvtapasvi) [inline src/DroneSafetyApi/Models/HeatmapsResponse.cs]: Perhaps should say in here that length(Sources) = NumSources = length(Heatmaps), and the keys of heatmaps correspond exactly the the elements in Sources [Mention invariants]

File: src/DroneSafetyApi/Services/HazardsToHeatmapsResponseNoCompositionSendAllSources.cs [hunk 10]
```diff
@@ -4,14 +4,28 @@
 
 namespace DroneSafetyApi.Services
 {
+    /// <summary>
+    /// The HazardsToHeatmapsResponseNoCompositionSendAllSources class is a service to generate <see cref="HeatmapsResponse"/>s
```
Comment (dhruvtapasvi) [inline src/DroneSafetyApi/Services/HazardsToHeatmapsResponseNoCompositionSendAllSources.cs]: Perhaps explain the name: why is it no composition? why is it send all sources?

File: src/DroneSafetyApi/Services/IHazardsToHeatmapsResponse.cs [hunk 11]
```diff
@@ -4,8 +4,21 @@
 
 namespace DroneSafetyApi.Services
 {
+    /// <summary>
+    /// The IHazardsToHeatmapsResponse interface is capable of generating <see cref="HeatmapsResponse"/>s
```
Comment (dhruvtapasvi) [inline src/DroneSafetyApi/Services/IHazardsToHeatmapsResponse.cs:8]: Say that this is the most general interface, which must be implemented by any strategy created a HeatmapsResponse from a Hazard list

File: src/DroneSafetyApi/Services/IHeatmapFactory.cs [hunk 12]
```diff
@@ -6,8 +6,17 @@
 
 namespace DroneSafetyApi.Services
 {
+    /// <summary>
+    /// The IHeatmapFactory interface is capable of generating new <see cref="IHeatmap"/>s
```
Comment (dhruvtapasvi) [inline src/DroneSafetyApi/Services/IHeatmapFactory.cs]: Why should we use a factory as opposed to using new? Say that it makes the code more testable, by depending on dependency injection. Say that this is the Factory design pattern (or mention that this class is a factory for IHeatmap instances)

**Review by dhruvtapasvi — COMMENTED** [review]
Very good overall, should just perhaps be more specific with one or two class level documentation points as I've noted, that's all.

File: src/DroneSafetyApi/Data/DbOptions.cs (hunk 4 above)
Comment (ranzmigrod) [reply src/DroneSafetyApi/Data/DbOptions.cs:14]: I have done it for all the properties so wanted to be consistent, I can take it away if you want

Comment (dhruvtapasvi) [reply src/DroneSafetyApi/Data/DbOptions.cs:14]: No problem doesn't really matter

File: src/DroneSafetyApi/Services/IHazardsToHeatmapsResponse.cs (hunk 11 above)
Comment (ranzmigrod) [reply src/DroneSafetyApi/Services/IHazardsToHeatmapsResponse.cs:8]: Don't think this is necessary as it is an interface so its pretty clear

File: src/DroneSafetyApi/Services/IHeatmapFactory.cs (hunk 12 above)
Comment (ranzmigrod) [reply src/DroneSafetyApi/Services/IHeatmapFactory.cs]: I will mention the it is a factory design pattern, the more testable shouldn't go in this documentation

**Pull request merged**